subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformInfoExecutor.java

```diff
@@ -18,13 +18,38 @@
 
 import org.gradle.execution.taskgraph.WorkInfo;
 import org.gradle.execution.taskgraph.WorkInfoExecutor;
+import org.gradle.internal.operations.BuildOperationCategory;
+import org.gradle.internal.operations.BuildOperationContext;
+import org.gradle.internal.operations.BuildOperationDescriptor;
+import org.gradle.internal.operations.BuildOperationExecutor;
+import org.gradle.internal.operations.RunnableBuildOperation;
 
 public class TransformInfoExecutor implements WorkInfoExecutor {
+    private final BuildOperationExecutor buildOperationExecutor;
+
+    public TransformInfoExecutor(BuildOperationExecutor buildOperationExecutor) {
+        this.buildOperationExecutor = buildOperationExecutor;
+    }
+
     @Override
     public boolean execute(WorkInfo work) {
         if (work instanceof TransformInfo) {
-            TransformInfo transform = (TransformInfo) work;
-            transform.execute();
+            final TransformInfo transform = (TransformInfo) work;
+            buildOperationExecutor.run(new RunnableBuildOperation() {
+                @Override
+                public void run(BuildOperationContext context) {
+                    transform.execute();
+                }
+
+                @Override
+                public BuildOperationDescriptor.Builder description() {
+                    String displayName = "Transform " + transform.toString();
```

Comment:
For combined transforms the arrow seems to be the wrong way: 
```
> Transform FindRelocatedClasses -> ShadeClassesTransform -> MinifyTransform on groovy-all.jar (org.codehaus.groovy:groovy-all:2.4.12)
```

The data flow is in the opposite direction of the arrows. Do you think the other arrow direction would be better?